File: go/iris-squirrel-example/api/jutsu/jutsu.go
package jutsu

import (
	"iris-squirrel-example/providers"
	"iris-squirrel-example/types"

	iris "github.com/kataras/iris/v12"
)

func RegisterJutsu(app *iris.Application, providers *providers.Providers) {
	r := app.Party("/api/jutsu")
	{
		// Get jutsu
		r.Get("/{id}", func(c iris.Context) {
			id := c.Params().Get("id")
			jutsu, err := providers.PGDAL.GetJutsu(id)
			if err != nil {
				c.StatusCode(iris.StatusInternalServerError)
				return
			}
			c.JSON(jutsu)
		})

		// Insert jutsu
		r.Post("/", func(c iris.Context) {
			var jutsuNew types.JutsuNew
			err := c.ReadJSON(&jutsuNew)
			if err != nil {
				c.StatusCode(iris.StatusBadRequest)
				return
			}
			jutsu, err := providers.PGDAL.CreateJutsu(jutsuNew)
			if err != nil {
				c.StatusCode(iris.StatusInternalServerError)
				return
			}
			c.JSON(jutsu)
		})

		// Update jutsu
		r.Put("/{id}", func(c iris.Context) {
			id := c.Params().Get("id")
			var jutsuUpdates types.JutsuNew
			err := c.ReadJSON(&jutsuUpdates)
			if err != nil {
				c.StatusCode(iris.StatusBadRequest)
				return
			}
			jutsu, err := providers.PGDAL.UpdateJutsu(id, jutsuUpdates)
			if err != nil {
				c.StatusCode(iris.StatusInternalServerError)
				return
			}
			c.JSON(jutsu)
		})

		// Delete Jutsu
		r.Delete("/{id}", func(c iris.Context) {
			id := c.Params().Get("id")
			jutsu, err := providers.PGDAL.DeleteJutsu(id)
			if err != nil {
				c.StatusCode(iris.StatusInternalServerError)
				return
			}
			c.JSON(jutsu)
		})
	}
}

File: py/fastapi-pypika-example/src/api/jutsu/jutsu.py
from uuid import UUID
from fastapi import Depends, Response, APIRouter
from ...db.providers import get_providers, Providers
from ...types.types import JutsuNew, JutsuUpdates

router = APIRouter()


@router.get("/{id}")
async def get_jutsu(
    response: Response,
    id: str,
    providers: Providers = Depends(get_providers)
):
    """Get jutsu"""
    if not isinstance(id, str):
        response.status_code = 400
        return {}
    try:
        jutsu = await providers.pgdal.get_jutsu(UUID(id))
        response.status_code = 200
        return jutsu
    except Exception:
        response.status_code = 500
        return {}


@router.post("/")
async def create_jutsu(
    response: Response,
    jutsu_new: JutsuNew,
    providers: Providers = Depends(get_providers)
):
    """Insert new jutsu"""
    try:
        jutsu = await providers.pgdal.create_jutsu(jutsu_new)
        response.status_code = 200
        return jutsu
    except Exception:
        response.status_code = 500
        return {}


@router.put("/{id}")
async def update_jutsu(
    response: Response,
    id: str,
    jutsu_updates: JutsuUpdates,
    providers: Providers = Depends(get_providers)
):
    """Update existing jutsu"""
    if not isinstance(id, str):
        response.status_code = 400
        return {}
    try:
        jutsu = await providers.pgdal.update_jutsu(UUID(id), jutsu_updates)
        response.status_code = 200
        return jutsu
    except Exception:
        response.status_code = 500
        return {}


@router.delete("/{id}")
async def delete_jutsu(
    response: Response,
    id: str,
    providers: Providers = Depends(get_providers)
):
    """Delete existing jutsu"""
    if not isinstance(id, str):
        response.status_code = 400
        return {}
    try:
        jutsu = await providers.pgdal.delete_jutsu(UUID(id))
        response.status_code = 200
        return jutsu
    except Exception:
        response.status_code = 500
        return {}
